go-runtime/internal/api.go

```diff
@@ -56,6 +58,15 @@ func ContextWithCallMetadata(ctx context.Context, metadata map[MetadataKey]strin
 	return context.WithValue(ctx, callMetadataKey{}, metadata)
 }
 
+// MaybeCallMetadataFromContext returns the call metadata from the context if it exists.
+func MaybeCallMetadataFromContext(ctx context.Context) optional.Option[map[MetadataKey]string] {
+	metadata, ok := ctx.Value(callMetadataKey{}).(map[MetadataKey]string)
+	if !ok {
+		return optional.None[map[MetadataKey]string]()
+	}
+	return optional.Some(metadata)
```

Comment:
There's a helper function for this!

```suggestion
	metadata, ok := ctx.Value(callMetadataKey{}).(map[MetadataKey]string)
	return optional.From(metadata, ok)
```

Reply:
oooh excellent!!!